### PR TITLE
Upgrade TypeSpec to 0.43.0

### PR DESCRIPTION
**Pull request by chunyu3**
# Description
Fix https://github.com/Azure/autorest.csharp/issues/3272

- upgrade typespec library to 0.43.0
- resolve the breaking changes
- update cadl-ranch projects

# Checklist

To ensure a quick review and merge, please ensure:
- [ ] The PR has a understandable title and description explaining the _why_ and _what_.
- [ ] The PR is opened in draft if not ready for review yet.
   - If opened in draft, please allocate sufficient time (24 hours) after moving out of draft for review
- [ ] The branch is recent enough to not have merge conflicts upon creation.

# Ready to Land?
- [ ] Build is completely green
   - Submissions with test failures require tracking issue and approval of a CODEOWNER
- [ ] At least one +1 review by a CODEOWNER
- [ ] All -1 reviews are confirmed resolved by the reviewer 
   - Override/Marking reviews stale must be discussed with CODEOWNERS first

### CODE REVIEW

File: test/CadlRanchProjects.Tests/CollectionFormat.cs [hunk 1]
```diff
@@ -16,32 +16,24 @@ public class CollectionFormatTests : CadlRanchTestBase
         public Task CollectionFormat_testMulti() => Test(async (host) =>
```
Comment (pshao25) [inline test/CadlRanchProjects.Tests/CollectionFormat.cs:16]: Rename to `Parameters_CollectionFormat_Query_multi`

File: test/CadlRanchProjects.Tests/CollectionFormat.cs [hunk 2]
```diff
@@ -16,32 +16,24 @@ public class CollectionFormatTests : CadlRanchTestBase
...
         {
             List<string> colors = new List<string>() { "blue", "red", "green" };
-            Response response = await new CollectionFormatClient(host, null).TestMultiAsync(colors);
-            Assert.AreEqual(200, response.Status);
+            Response response = await new CollectionFormatClient(host, null).GetQueryClient().MultiAsync(colors);
+            Assert.AreEqual(204, response.Status);
         });
 
         [Test]
         public Task CollectionFormat_testCsv() => Test(async (host) =>
```
Comment (pshao25) [inline test/CadlRanchProjects.Tests/CollectionFormat.cs:24]: Rename to `Parameters_CollectionFormat_Query_csv`

File: test/CadlRanchProjects.Tests/CollectionFormat.cs [hunk 3]
```diff
@@ -16,32 +16,24 @@ public class CollectionFormatTests : CadlRanchTestBase
...
         {
             List<string> colors = new List<string>() { "blue", "red", "green" };
-            Response response = await new CollectionFormatClient(host, null).TestCsvAsync(colors);
-            Assert.AreEqual(200, response.Status);
+            Response response = await new CollectionFormatClient(host, null).GetQueryClient().CsvAsync(colors);
+            Assert.AreEqual(204, response.Status);
         });
 
         [Test]
         public Task CollectionFormat_testCsvHeader() => Test(async (host) =>
```
Comment (pshao25) [inline test/CadlRanchProjects.Tests/CollectionFormat.cs:32]: Rename to `Parameters_CollectionFormat_Header_csv`

**Review by pshao25 — APPROVED** [review]

File: src/CADL.Extension/Emitter.Csharp/package.json [hunk 4]
```diff
@@ -50,17 +50,17 @@
...
+        "@typespec/compiler": "0.43.0",
         "@typespec/eslint-config-typespec": "0.6.0",
-        "@typespec/eslint-plugin": "0.42.0",
-        "@typespec/http": "0.42.0",
-        "@typespec/library-linter": "0.42.0",
-        "@typespec/openapi": "0.42.0",
-        "@typespec/rest": "0.42.0",
-        "@typespec/versioning": "0.42.0",
+        "@typespec/eslint-plugin": "0.43.0",
+        "@typespec/http": "0.43.0",
```
Comment (lirenhe) [inline src/CADL.Extension/Emitter.Csharp/package.json:59]: Can we update this to 0.43.1 which contains the bug fix for collection format.

**Comment by lirenhe** [comment]
@chunyu3 , do you have the SDK Regen PR?  I think all typespec files are updated so it should not block the regen.

File: src/CADL.Extension/Emitter.Csharp/src/lib/model.ts [hunk 5]
```diff
@@ -133,7 +133,9 @@ function getCSharpInputTypeKindByIntrinsicModelName(
             return InputTypeKind.Boolean;
         case "plainDate":
             return InputTypeKind.Date;
-        case "zonedDateTime":
+        case "utcDateTime":
+            return InputTypeKind.DateTime;
+        case "offsetDateTime":
```
Comment (lirenhe) [inline src/CADL.Extension/Emitter.Csharp/src/lib/model.ts:138]: Just curious what change needed for the serialization and deserialization for those new datetime types.
As this should not be a breaking change, we could decide that later.

**Comment by chunyu3** [comment]
> @chunyu3 , do you have the SDK Regen PR? I think all typespec files are updated so it should not block the regen.

@lirenhe regen PR is generated.

File: src/CADL.Extension/Emitter.Csharp/src/lib/model.ts (hunk 5 above)
Comment (lirenhe) [reply src/CADL.Extension/Emitter.Csharp/src/lib/model.ts:138]: Discussed with Mike,  no change needed in codegen for serialization and deserialization

**Review by lirenhe — APPROVED** [review]

File: src/CADL.Extension/Emitter.Csharp/package.json (hunk 4 above)
Comment (chunyu3) [reply src/CADL.Extension/Emitter.Csharp/package.json]: updated

**Comment by m-nash** [comment]
@chunyu3 this PR looks good but I am holding off merging it because the regen PR is having issues with finding the alpha package in the public feed.  I reached out to @hallipr but we weren't able to fully resolve the problem.

I believe its fine to merge this in as the resulting regen is visible and we can update to the beta packages once this is merged but I will let you do that tonight.

**Review by m-nash — APPROVED** [review]

**Pull request merged**